src/trio/testing/_raises_group.py

```diff
@@ -383,6 +384,15 @@ def matches(
         # maybe have a list of strings logging failed matches, that __exit__ can
         # recursively step through and print on a failing match.
         if not isinstance(exc_val, BaseExceptionGroup):
+            if not self.strict and len(self.expected_exceptions) == 1:
+                exp_exc = self.expected_exceptions[0]
+                if isinstance(exp_exc, Matcher) and exp_exc.matches(exc_val):
+                    return True
+                if isinstance(exp_exc, type) and isinstance(exc_val, exp_exc):
+                    return True
+            # Consider printing a helpful message on unwrapped exception, strict=False and
+            # `len(self.expected_exceptions) > 1`; since they might expect the exceptions
+            # to be 'or'd like with `pytest.raises`.
```

Comment:
if it's a valid usage, the message might be annoying or even interfere with tooling (where a project tries to keep test runs free of warning messages)

Reply:
It'd only be:
* if the code is `with RaisesGroups(excA, excB, strict=False): ...`
* and a single exception [that would match excA or excB] is raised

Either way your test case is failing, so would be similar to how `pytest.raises` tries to be helpful when it looks like a user has failed to `re.escape` a `match=` string.

In general I think the matching & construction of stuff like nested groups is complicated enough that the tool should be emitting when something is close to matching but doesn't. Also see
https://github.com/python-trio/trio/blob/62c0ea02e0bf98188544c6e0136caa6b46911b92/src/trio/testing/_raises_group.py#L383-L385